### ISSUE TITLE
SQLite errors after upgrade to 3.5 - CHECK constraint failed: sys_hwinfo

### ISSUE THREAD

**Issue by VAdamec** [wazuh/wazuh#1099]
Hi I'm getting this error after upgrade:

```
2018/08/14 01:32:58 wazuh-db: ERROR: at wdb_hardware_insert(): sqlite3_step(): CHECK constraint failed: sys_hwinfo
2018/08/14 01:32:58 wazuh-db: ERROR: Unable to update 'sys_hwinfo' table for agent '259'
2018/08/14 01:32:58 ossec-analysisd: ERROR: at sc_send_db(): received: 'err Cannot save HW information.'
2018/08/14 01:32:58 wazuh-db: ERROR: at wdb_hardware_insert(): sqlite3_step(): CHECK constraint failed: sys_hwinfo
2018/08/14 01:32:58 wazuh-db: ERROR: Unable to update 'sys_hwinfo' table for agent '266'
2018/08/14 01:32:58 ossec-analysisd: ERROR: at sc_send_db(): received: 'err Cannot save HW information.'
2018/08/14 01:33:04 wazuh-db: ERROR: at wdb_hardware_insert(): sqlite3_step(): CHECK constraint failed: sys_hwinfo
2018/08/14 01:33:04 wazuh-db: ERROR: Unable to update 'sys_hwinfo' table for agent '275'
2018/08/14 01:33:04 ossec-analysisd: ERROR: at sc_send_db(): received: 'err Cannot save HW information.'
```

Oracle Linux 7.5
wazuh-api-3.5.0-1.x86_64
wazuh-manager-3.5.0-1.x86_64

Did I miss something in upgrade process ?

**Comment by chemamartinez** [comment]
Hi @VAdamec,

The upgrade process should delete the complete databases to avoid incompatibilities between Wazuh versions when a new scan arrives. The error you are getting is about one of the value checks that SQLite makes before storing the information in the table (for example not receive a negative number for the number of cores of the processor). 

It would be very useful to be able to see the query that causes that error to determine which value fails. To do this you could follow the following steps:

1. Set the `wazuh-db` module on debug mode:
```
# echo "wazuh_db.debug=2" >> /var/ossec/etc/local_internal_options.conf
```

2. Restart one of the affected agents to trigger a new inventory scan (this can be done from the manager side):
```
# /var/ossec/bin/agent_control -R -u <agent_id>
```

3. Print retrieved error and its related query (is shown by a debug message):
```
# cat /var/ossec/logs/ossec.log | grep -C 5 "CHECK constraint failed"
```

You should get something like the following logs:
```
2018/08/14 12:52:58 wazuh-db: DEBUG: Executing query: hardware save 1179689414|2018/08/14 12:52:58|341341AE13|Intel(R) Core(TM) i7-7660U CPU @ 2.50GHz|1|2496.001000|4015688|1193216|71
2018/08/14 01:33:04 wazuh-db: ERROR: at wdb_hardware_insert(): sqlite3_step(): CHECK constraint failed: sys_hwinfo
2018/08/14 01:33:04 wazuh-db: ERROR: Unable to update 'sys_hwinfo' table for agent '275'
2018/08/14 01:33:04 ossec-analysisd: ERROR: at sc_send_db(): received: 'err Cannot save HW information.'
```

I'll be waiting for your answer.

Thank you,
Chema.

**Comment by chemamartinez** [comment]
Hi @VAdamec , 

I have found the issue root and it has been fixed at this PR https://github.com/wazuh/wazuh/pull/1124, it was caused by the calculation of a new field included in v.3.5.0. 
The fix will be available in the next released version, thank you for the feedback. 

Meanwhile, I would recommend you to disable the hardware scan for the issued agents, sorry for the inconvenience.

Regards,
Chema.
